var/spack/repos/builtin/packages/amber/package.py

```diff
@@ -17,63 +18,177 @@ class Amber(Package, CudaPackage):
+        # The resource command does not allow us to expand the package in the
+        # root stage folder as required, as it already contains files. Here we
+        # install AmberTools where it should be, which results in 3 copies of
+        # the  ambertools (~9 GB). This has to be improved in the future.
+        install_tree('ambertools_tmpdir', '.')
+        shutil.rmtree(join_path(self.stage.source_path, 'ambertools_tmpdir'))
+
+        # Select compiler style
+        if self.spec.satisfies('+cray'):
+            compiler = 'cray'
+        elif self.spec.satisfies('%gcc'):
+            compiler = 'gnu'
+        elif self.spec.satisfies('%intel'):
+            compiler = 'intel'
+        elif self.spec.satisfies('%pgi'):
+            compiler = 'pgi'
+        elif self.spec.satisfies('%clang'):
+            compiler = 'clang'
+        else:
+            raise InstallError('Unknown compiler, exiting!!!')
+
+        # Base configuration
         conf = Executable('./configure')
-
-        conf(*(base_args + ['gnu']), extra_env=configure_env)
-        make('install', extra_env=configure_env)
-
-        if '+mpi' in spec:
-            conf(*(base_args + ['-mpi', 'gnu']), extra_env=configure_env)
-            make('install', extra_env=configure_env)
-
-        if '+cuda' in spec:
-            conf(*(base_args + ['-cuda', 'gnu']), extra_env=configure_env)
-            make('install', extra_env=configure_env)
+        base_args = ['--skip-python',
+                     '--with-netcdf', self.spec['netcdf-fortran'].prefix,
+                     ]
+        if self.spec.satisfies('~x11'):
+            base_args += ['-noX11']
+
+        # Update the sources: Apply all upstream patches
+        if self.spec.satisfies('+update'):
+            update = Executable('./update_amber')
+            update(*(['--update']))
+        else:
+            base_args += ['--no-updates']
+
+        # Single core
+        conf(*(base_args + [compiler]))
+        make('install')
+
+        # CUDA
+        if self.spec.satisfies('+cuda'):
+            conf(*(base_args + ['-cuda', compiler]))
+            make('install')
+
+        # MPI
+        if self.spec.satisfies('+mpi'):
+            conf(*(base_args + ['-mpi', compiler]))
+            make('install')
+
+        # Openmp
+        if self.spec.satisfies('+openmp'):
+            make('clean')
+            conf(*(base_args + ['-openmp', compiler]))
+            make('openmp')
+
+        # CUDA + MPI
+        if self.spec.satisfies('+cuda') and self.spec.satisfies('+mpi'):
+            make('clean')
+            conf(*(base_args + ['-cuda', '-mpi', compiler]))
+            make('install')
 
         # just install everything that was built
         install_tree('.', prefix)
+
+    def setup_run_environment(self, env):
+        env.set('AMBER_PREFIX', self.prefix)
+        env.set('AMBERHOME', self.prefix)
+        env.prepend_path('PATH', self.prefix.bin)
+        env.prepend_path('LD_LIBRARY_PATH', self.prefix.lib)
```

Comment:
These two aren't needed, they are automatically added to the module file.

Reply:
All suggestions accepeted and commited.